daliuge-engine/dlg/drop.py

```diff
@@ -1715,6 +1717,24 @@ def initialize(self, **kwargs):
         super(BarrierAppDROP, self).initialize(**kwargs)
 
 
+class PlasmaDROP(AbstractDROP):
+    '''
+    A DROP that points to data stored in a Plasma Store
+    '''
+    object_id = dlg_string_param('object_id', plasma.ObjectID.from_random().binary())
```

Comment:
Maybe default to `None` and then set to `self.uid` (or rather, a 20-bytes-long version of `self.uid` -- substring, hash, any other method) during `initialize`? That would also remove the need to import `pyarrow` at the top level.

Reply:
This also currently yields a single, random, default object ID for all instances of `PlasmaDROP`, right?

Reply:
Yes I think I will set it to `None'  then defer the object id generation it to an external source. 

I think Andreas is happy to have pyarrow as a base package but i need to check.